kernel/pmu_x86/src/lib.rs

```diff
@@ -701,7 +699,8 @@ pub fn start_samples(event_type: EventType, event_per_sample: u32, task_id: Opti
     // This check can never trigger since `event_per_sample` is a `u32`
     // and is therefore by definition in the range `u32::MIN..=u32::MAX`.
     // We'll check anyways, just in case `event_per_sample`'s type is changed.
-    if u32::try_from(event_per_sample).is_ok() {
+    
+    if (u32::MIN..=u32::MAX).contains(&event_per_sample) {
```

Comment:
kindly keep the iteration of `u32::try_from()`. That is the _clearest_ and most correct semantic meaning of what we want to achieve here, and even when we change the type of `event_per_sample`, that code will continue to work. This change would not.

Reply:
Actually, the condition needs to be:
```suggestion
    if u32::try_from(event_per_sample).is_err() {
```
emphasis on the `is_err()`